SDWebImage/SDDiskCache.h

```diff
@@ -104,4 +104,15 @@
 
 - (nonnull instancetype)init NS_UNAVAILABLE;
 
+/**
+ Move the cache directory from old location to new location, the old location will be removed after finish.
+ If the old location does not exist, does nothing.
+ If the new location does not exist, only do a movement of directory.
+ If the new location does exist, will move and merge the files from old location.
+
+ @param srcPath old location of cache directory
+ @param dstPath new location of cache directory
+ */
+- (void)moveCacheDirectoryFromPath:(nonnull NSString *)srcPath toPath:(nonnull NSString *)dstPath;
```

Comment:
@dreampiggy I updated the `PR` I make. The good thing is we reached the decision for the most of the code. The only difference is wether we should make this method public. 

If you decide to make it public and make it for common use, I think maybe can make some changes:
1. What if `dstPath` not exist, and `dstPath` has non-existent parent directories, your code can't handle correctly.
2. If `dstPath` is not a directory but exist, should we need to delete the file directly? maybe it needs to think a bit more.
3.  For the merge operation, what if the `dstFile` exist? for the current code, it would raise the error and skip this movement of file, it's the user want? Can we provide overwrite operation?
4. Should we need to provide any feedback, like move successful or error parameter to the user?

Otherwise, if you think we can make it private, I think my `PR` can handle very well, we only handle our own default cache migration.
At last, it's up to you to make the decision, I'm ok for these two `PR`s.

Reply:
1. When `dstPath` does not exist, we do a movement of directory, which is more easier.
2. When `dstPath` is a file, we remove it and fallthrough to the 1. case above.
3. When merge exist, this case only happened if both `srcPath` && `dstPath` are directory, so this will not raise error, until you pass a `dstPath` which is not writeable (Such as `/System/Library`)
4. Previously I think we can put a error pointer. I don't know whether user want to know the erorr, if they want, they can use the `NSFileManagerDelegate` to handle the error. So I don't add one arg here. This can be still in discussion.
5. This is just a movement of `SDDiskCahce` disk cache. It's not a helper method or util. For example, maybe someday we change the implementration to also rename something during cache path movement. It's possible. So it's better not to treate this as a common method

Reply:
@dreampiggy 
1. We can't just use move operation, like my comment, if dstPath's parent directory don't exist, it would not do success.
3. When merge exist, except not writable, if dstPath directory has exist the same name file which also exist in oldPath, error would raise.

Yeah, it's also what I wanted, it's not a common use method, only used in cache migration for ourself.

Reply:
Correct. I'll provide a fix for the 1. case...

So the 2. one, it seems that during merge we ommit the error and continue, isn't it adoptable? And since you can always get the error delegate method from NSFileManager, maybe keep this is OK